### ISSUE TITLE
handling the different sized labeled and unlabeled loaders

### ISSUE THREAD

**Issue by anu43** [wohlert/semi-supervised-pytorch#33]
Hello,

I wonder about the idea behind this approach. Wouldn't repeating labeled data cause overfitting in the training?

https://github.com/wohlert/semi-supervised-pytorch/blob/7cde4959468d271552febdcfed5e1cfae9857613/examples/mnist_sslvae.py#L92

Thanks in advance!

**Comment by wohlert** [comment]
These kinds of models have a very hard time overfitting since the labelled data is so small. Here is the explanation: The gradient is averaged over a batch of labelled and unlabelled data. This gradient will point towards the labelled data, but will be augmented by the noise that the unlabelled data induces. Since the labelled data is paired with a new set of unlabelled data at each batch the gradient points towards different directions at each batch. This acts as a natural regulariser preventing overfitting.

 If you are worried about this you can set the alpha parameter lower such that the gradient from labelled data effectively vanishes.
